### PR TITLE
Model training overhaul: SNGP model, uncertainty visualization, and custom embedding support

### PR DESCRIPTION
**Pull request by ykevu**
## What
* Custom embeddings can be imported for visualization and training by including embeddings.json in an uploaded zip file
* Various UI tweaks and fixes
* Implement tabs for different visualizations in the training modal, including new training set histogram
* Implement SNGP model using custom-written layers and functions adapted from Python Tensorflow
* Uncertainty for predictions can be visualized and calculated for model inference

### CODE REVIEW

**Comment by ykevu** [comment]
@rossbar Sorry for the mess of commits, but this should be ready now. The actual changes span from [d2e71c3](https://github.com/vanvalenlab/deepcell-label/pull/436/commits/d2e71c3641d2bae70f751a11af1a660e968a58c4) to [4bfd476](https://github.com/vanvalenlab/deepcell-label/pull/436/commits/4bfd476be73685e445be385702f7c6bee92344c5), then [46cc3ce](https://github.com/vanvalenlab/deepcell-label/pull/436/commits/46cc3ce31e64291a9ed3d2e262b20c0b9f9e4e17) onwards.

Can probably have things like testing, fixes for bugs that crop up, and any additional features rolled into separate / later PRs

File: frontend/src/Project/EditControls/CellTypeControls/TrainingUI/EmbeddingPlot.js [hunk 1]
```diff
@@ -17,8 +40,27 @@ function EmbeddingPlot({ embedding }) {
...
+  const colorScale = [
+    ['0', 'rgb(49,54,149)'],
+    ['0.111111111111', 'rgb(69,117,180)'],
+    ['0.222222222222', 'rgb(116,173,209)'],
+    ['0.333333333333', 'rgb(171,217,233)'],
+    ['0.444444444444', 'rgb(224,243,248)'],
+    ['0.555555555556', 'rgb(254,224,144)'],
+    ['0.666666666667', 'rgb(253,174,97)'],
+    ['0.777777777778', 'rgb(244,109,67)'],
+    ['0.888888888889', 'rgb(215,48,39)'],
```
Comment (rossbar) [inline frontend/src/Project/EditControls/CellTypeControls/TrainingUI/EmbeddingPlot.js:61]: The hard-coding of `N/9` as strings seems strange, but if the underlying machinery expects strings and there is no better way to do formatted strings in javascript (i.e. something like what `f"(1 / 9)"` would do in Python) then +1.

File: frontend/src/Project/service/labels/cellTypesMachine.js [hunk 2]
```diff
@@ -323,7 +323,7 @@ const createCellTypesMachine = ({ eventBuses, undoRef }) =>
           const pred = evt.predictions;
           for (let cell in pred) {
             cellTypes = cellTypes.map((cellType) =>
-              cellType.id === pred[cell] + 1 && !cellType.cells.includes(cell)
+              cellType.id === pred[cell] + 1 && !cellType.cells.includes(cell) && cell > 0
```
Comment (rossbar) [inline frontend/src/Project/service/labels/cellTypesMachine.js:326]: What is the new `cell > 0` check catching? Not a blocker, I'm just curious!

**Review by rossbar — APPROVED** [review]
LGTM - I can't give any constructive feedback on the tfjs component, which is the meat of this PR. In principle it'd be great if we could import & extend the various tfjs layers and associated utils rather than vendoring them, but I don't know how to go about doing so and don't think it should be a blocker.

File: frontend/src/Project/EditControls/CellTypeControls/TrainingUI/EmbeddingPlot.js (hunk 1 above)
Comment (ykevu) [reply frontend/src/Project/EditControls/CellTypeControls/TrainingUI/EmbeddingPlot.js:61]: It's how they did it in the plotly documentation but I think I can just change it to (1 / 9).toFixed(12) for a 12 decimal place string.

File: frontend/src/Project/service/labels/cellTypesMachine.js (hunk 2 above)
Comment (ykevu) [reply frontend/src/Project/service/labels/cellTypesMachine.js:326]: Cell 0 is the background, so the background ends up getting labeled by a cell type without it
